src/main/kotlin/io/zerobase/smarttracing/resources/Amazon.kt

```diff
@@ -0,0 +1,179 @@
+        if (config.from != "") {
+            val message = MimeMessage(session)
+
+            message.setSubject(subject, "UTF-8")
+            message.setFrom(InternetAddress(config.from))
+            message.setRecipients(Message.RecipientType.TO, InternetAddress.parse(to))
+
+            val attachment = MimeBodyPart()
+            val html = MimeBodyPart()
+
+            html.setContent(html_string, "text/html; charset=UTF-8")
+            // TODO: Make the qr code generator
+            //attachment.setDataHandler(DataHandler(createQR(url)))
+            //attachment.setFileName("scannable-qr.pdf")
+
+            val msg = MimeMultipart("mixed")
+
+            message.setContent(msg)
+            msg.addBodyPart(html)
+            msg.addBodyPart(attachment)
+
+            val outStream = ByteArrayOutputStream()
+
+            message.writeTo(outStream)
+
+            val arr = outStream.toByteArray()
+
+            val data = SdkBytes.fromByteArray(arr)
+            val rawMessage = RawMessage.builder().data(data).build()
+            val rawEmailRequest = SendRawEmailRequest.builder().rawMessage(rawMessage).build()
+
+            client.sendRawEmail(rawEmailRequest)
+        }
+    }
+
+    /**
+     * Sends thank you for joining email to a user
+     *
+     * @param to address to send to
+     *
+     * @throw AddressException when the address is incorrect
+     * @throw MessagingException when we cannot message
+     * @throw IOException input output exception
+     */
+    fun thanksForJoiningUser(to: String) {
+        sendNormal("Thanks for Opting In", "You have opted into the zerobase program", to)
+    }
+
+    /**
+     * Sends thank you for joining email to an organization
+     *
+     * @param to address to send to
+     *
+     * @throw AddressException when the address is incorrect
+     * @throw MessagingException when we cannot message
+     * @throw IOException input output exception
+     */
+    fun thanksForJoiningOrg(to: String) {
+        sendNormal("Thanks for Opting In", "You have opted into the zerobase program, please place some scannables", to)
+    }
+
+    /**
+     * Sends thank you for deletine email to a user
+     *
+     * @param to address to send to
+     *
+     * @throw AddressException when the address is incorrect
+     * @throw MessagingException when we cannot message
+     * @throw IOException input output exception
+     */
+    fun thanksForDeletingUser(to: String) {
+        sendNormal("Thanks for Deleting", "We are sorry to hear you go, if you could we would like you to fill out your thoughts on us", to)
+    }
+
+    /**
+     * Sends thank you for joining email to an organization
+     *
+     * @param to address to send to
+     *
+     * @throw AddressException when the address is incorrect
+     * @throw MessagingException when we cannot message
+     * @throw IOException input output exception
+     */
+    fun scannableOrgEmail(to: String, url: String) {
```

Comment:
none of these function names describe the action being taken. on top of that, the contents shouldn't be in the email abstraction. at the very least, they should be templates stored as resources.

Reply:
what names do you suggest? i am bad at naming, i am moving the templates into resources now

Reply:
for now, let's keep the email abstraction simple: it takes an address, a body, and an optional attachment. we can make another `NotificationManager` class to abstract whether a user is notified by email or sms or push notification.

Reply:
alright will push that soon